cmd/kind/kind.go

```diff
@@ -76,14 +90,13 @@ func NewCommand() *cobra.Command {
 }
 
 func runE(flags *Flags) error {
-	level := defaultLevel
-	parsed, err := log.ParseLevel(flags.LogLevel)
-	if err != nil {
-		log.Warnf("Invalid log level '%s', defaulting to '%s'", flags.LogLevel, level)
+	// TODO: handle --loglevel which previously supported:
+	// panic, fatal, error, warning, info, debug, trace
```

Comment:
Not for or against this, but when is this TODO going to be done given we intend to remove this flag anyway? Seems like it either happens now, or doesn't happen (because by the time it comes to implementing it, we'll have removed the flag anyway)

Reply:
i think we probably need at least one release where the flag is still accepted but either just triggers a warning or triggers a warning and guesses at setting a roughly equivalent verbosity (like, if you were setting DEBUG, maybe v3, if you were setting TRACE, maybe maximum verbosity)

then the _next_ release would drop it entirely.

otherwise we risk surprise breaking a bunch of CI. OTOH we _could_ do that since we're in alpha, but it seems maybe better to put it in the release notes and print a warning about migrating?

this is TODO because I wasn't sure what the mapping should be yet.

Reply:
I think the main thing we want to not break immediately is all the CI setups enabling higher levels to get more debug info.

If we want to leave support in the next release then we should:

Detect if the flag was set, if so:
- print a deprecation warning
- if -v was NOT set, map trace to max int32, debug to 3, and anything else just to 0

But I can also see an argument for just breaking it and moving on, we already don't guarantee that arbitrary kind commits are ready for usage, and the release notes will contain a breaking change notice for this in the next release...

Reply:
I like the idea of not breaking the CLI for a release or so, but IMO just writing "WARNING: THE OPTION YOU SPECIFIED ACTUALLY DOESN'T DO WHAT YOU EXPECT ANYMORE, TRY USING -v".

If we don't implement the mapping logic in this PR, I think we're unlikely to implement it before we just remove the flag entirely

Reply:
I'll add a stab at this in the morning. I don't think it will be much work, IIRC it's possible in pflag to tell if a flag was specified or not as well 😅

Reply:
done